### PR TITLE
test ios accessibility bridge

### PR DESCRIPTION
**Pull request by cyanglaz**
This adds a scenario app test to test if the accessibility bridge is building the accessibility tree correctly on iOS.

It will require a dedicated devicelab ios device to run with the voiceover on. 

Fixes: https://github.com/flutter/flutter/issues/42669

Related: https://github.com/flutter/flutter/issues/42270

### CODE REVIEW

File: testing/scenario_app/ios/Scenarios/Scenarios/AppDelegate.m [hunk 1]
```diff
@@ -53,6 +53,23 @@ - (BOOL)application:(UIApplication*)application
...
+    // [engine.binaryMessenger
+    //  setMessageHandlerOnChannel:@"scenario_status"
+    //  binaryMessageHandler:^(NSData* _Nullable message, FlutterBinaryReply _Nonnull reply) {
+    //    [engine.binaryMessenger
+    //     sendOnChannel:@"set_scenario"
+    //     message:[@"accessibility" dataUsingEncoding:NSUTF8StringEncoding]];
+    //  }];
+    // [engine ensureSemanticsEnabled];
+    // self.window.rootViewController = flutterViewController;
+
```
Comment (cyanglaz) [inline testing/scenario_app/ios/Scenarios/Scenarios/AppDelegate.m]: commented out for faster tests. Will move this back when it is ready for review

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/AccessibilityUITests.m [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  XCUIElement *firstElement = [self.application.otherElements elementMatchingPredicate:predicate];
+  if (![firstElement waitForExistenceWithTimeout:30]) {
+    XCTFail(@"Failed due to not able to find any element with 30 seconds");
+  }
+
+  NSLog(@"%@", self.application.debugDescription);
+
+  XCUIElementQuery *query = self.application.otherElements;
+  XCUIElementQuery *allElementsWithA11yLabelQuery = [query matchingPredicate: predicate];
+  NSArray<XCUIElement *> *elements = allElementsWithA11yLabelQuery.allElementsBoundByAccessibilityElement;
```
Comment (goderbauer) [inline testing/scenario_app/ios/Scenarios/ScenariosUITests/AccessibilityUITests.m]: I am not familiar with this method and the documentation on it is spares... Does it actually guarantee that elements are returned in a11y traversal order?

Comment (cyanglaz) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/AccessibilityUITests.m]: As far as I know, the `query` always runs a BFS traversal in the a11y tree. So as long as we only fetch the direct children with the `predicate`, it should be returning the  child elements in the same order as what we set in foo.accessibilityElements.

Comment (cyanglaz) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/AccessibilityUITests.m]: actually it seems like I didn't push the latest commit. Now it should make more sense.

File: shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm [hunk 3]
```diff
@@ -235,6 +235,10 @@ - (NSString*)routeName {
   return nil;
 }
 
+- (NSString *)accessibilityIdentifier {
+  return [NSString stringWithFormat:@"%d", self.node.id];
```
Comment (goderbauer) [inline shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm]: Note: the node id is not stable. If you navigate away from a page and then back to the page the same elements there may have different nodes. Not sure if that matters...

Comment (cyanglaz) [reply shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm]: I added this only for testing(And add a comment on it). I started with testing based on label/value, but then I realized that we can't test pure container node (without a label/value) this way. If you can think of another way to verify a container node without a label/value, then we don't need this id.

**Comment by cbracken** [comment]
@cyanglaz is this still being worked on?

**Comment by cyanglaz** [comment]
> @cyanglaz is this still being worked on?

Yes. Having some other high priority things currently. Will circle back once other things are done.

**Comment by cbracken** [comment]
Since this is fairly unactionable and gets hit in triage each week, I'll close this for now. When this is ready for ready for review, please send a new PR and we'll review it quickly :)

**Pull request closed without merging**